File: API/IAMTtradingWA/Models/CustomsClearenceDetails/CustomsClearenceDetails.cs
namespace IAMTradingWA.Models
{
    using System;
    public class CustomsClearenceDetail
    {
        public int CCDetailId { get; set; }
        public int CClearenceId { get; set; }
        public int QDetailId { get; set; }
        public Nullable<decimal> CCDetailWeighFactor { get; set; }
        public Nullable<decimal> CCDetailFleteCCS { get; set; }
        public Nullable<decimal> CCDetailFleteMIA { get; set; }
        public Nullable<decimal> CCDetailOtherExpenses { get; set; }
        public Nullable<decimal> CCDetailAmountIDB { get; set; }
        public Nullable<decimal> CCDetailTariffPct { get; set; }
        public Nullable<decimal> CCDetailTariffAmount { get; set; }
        public Nullable<decimal> CCDetailTaxPct { get; set; }
        public Nullable<decimal> CCDetailTaxAmount { get; set; }
        public Nullable<decimal> CCDetailCustomsSrvPct { get; set; }
        public Nullable<decimal> CCDetailCustomsSrvAmount { get; set; }
        public Nullable<decimal> CCDetailGuarantee { get; set; }
        public Nullable<decimal> CCDetailProfitMargin { get; set; }
        public Nullable<decimal> CCDetailProfitAmount { get; set; }
        public Nullable<decimal> CCDetailCurrencyRate { get; set; }
        public Nullable<decimal> CCDetailLinePrice { get; set; }
        public virtual decimal QDetailQty { get; set; }
        public virtual decimal QDetailPrice { get; set; }
        public virtual decimal QDetailLineTotal { get; set; }
        public virtual Nullable<decimal> QDetailCost { get; set; }
        public virtual Nullable<decimal> QDetailLineCost { get; set; }
        public virtual int QHeaderId { get; set; }
        public virtual string ItemName { get; set; }
        public virtual string ItemNum { get; set; }
        public virtual string ItemGYCode { get; set; }
        public virtual Nullable<decimal> ItemTariffPct { get; set; }
        public virtual string ItemTariffCode { get; set; }
        public virtual Nullable<decimal> ItemLength { get; set; }
        public virtual Nullable<decimal> ItemHeight { get; set; }
        public virtual Nullable<decimal> ItemWidth { get; set; }
        public virtual Nullable<decimal> ItemVolume { get; set; }
        public virtual Nullable<decimal> ItemWeight { get; set; }
        public virtual string ItemNameSupplier { get; set; }
        public virtual string ItemNumSupplier { get; set; }
    }
}

File: WEB/app_backlog/app/model/CustomsClearenceDetails.js
Ext.define('IAMTrading.model.CustomsClearenceDetails', {
    extend: 'Ext.data.Model',
    idProperty: 'CCDetailId',
    autoLoad: false,

    fields: [
        { name:'CCDetailId', type:'int', defaultValue: null },
        { name:'CClearenceId', type:'int', defaultValue: null },
        { name:'QDetailId', type:'int', defaultValue: null },
        { name:'CCDetailWeighFactor', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailFleteCCS', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailFleteMIA', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailOtherExpenses', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailAmountIDB', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailTariffPct', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailTariffAmount', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailTaxPct', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailTaxAmount', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailCustomsSrvPct', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailCustomsSrvAmount', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailGuarantee', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailProfitMargin', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailProfitAmount', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailCurrencyRate', type:'float', useNull: true, defaultValue: null },
        { name:'CCDetailLinePrice', type:'float', useNull: true, defaultValue: null },
        { name:'QDetailQty', type:'float', defaultValue: null },
        { name:'QDetailPrice', type:'float', defaultValue: null },
        { name:'QDetailLineTotal', type:'float', defaultValue: null },
        { name:'QDetailCost', type:'float', useNull: true, defaultValue: null },
        { name:'QDetailLineCost', type:'float', useNull: true, defaultValue: null },
        { name:'QHeaderId', type:'int', defaultValue: null },
        { name:'ItemName', type:'string' },
        { name:'ItemNum', type:'string', useNull: true, defaultValue: null },
        { name:'ItemGYCode', type:'string', useNull: true, defaultValue: null },
        { name:'ItemTariffPct', type:'float', useNull: true, defaultValue: null },
        { name:'ItemTariffCode', type:'string', useNull: true, defaultValue: null },
        { name:'ItemLength', type:'float', useNull: true, defaultValue: null },
        { name:'ItemHeight', type:'float', useNull: true, defaultValue: null },
        { name:'ItemWidth', type:'float', useNull: true, defaultValue: null },
        { name:'ItemVolume', type:'float', useNull: true, defaultValue: null },
        { name:'ItemWeight', type:'float', useNull: true, defaultValue: null },
        { name:'ItemNameSupplier', type:'string', useNull: true, defaultValue: null },
        { name:'ItemNumSupplier', type:'string', useNull: true, defaultValue: null }
    ],

    proxy:{
        type:'rest',
        url: IAMTrading.GlobalSettings.webApiPath + 'CustomsClearenceDetails',
        headers: {
           'Authorization-Token': IAMTrading.GlobalSettings.tokenAuth
        },
        reader:{
            type:'json',
            root:'data',
            totalProperty: 'total',
            successProperty: 'success',
            messageProperty: 'message',
            idProperty: 'CCDetailId'
        },
        writer: {
            writeAllFields: false
        },
        afterRequest: function (request, success) {
            var silentMode = this.getSilentMode();

            if (request.action == 'read') {
                //this.readCallback(request);
            }
            else if (request.action == 'create') {
                if (!request.operation.success)
                {
                    Ext.popupMsg.msg("Warning", "Record was not saved!!!");
                    Ext.global.console.warn(request.proxy.reader.jsonData.message);
                } else {
                    if(!silentMode)
                    Ext.popupMsg.msg("Success","Saved Successfully!!!");
                }
            }
            else if (request.action == 'update') {
                if (!request.operation.success)
                {
                    Ext.popupMsg.msg("Warning", "Record was not saved!!!");
                    Ext.global.console.warn(request.proxy.reader.jsonData.message);
                } else {
                    if(!silentMode)
                    Ext.popupMsg.msg("Success","Saved Successfully!!!");
                }
            }
            else if (request.action == 'destroy') {
                if (!request.operation.success)
                {
                    Ext.popupMsg.msg("Warning", "Record was not Deleted");
                    //Ext.global.console.warn(request.proxy.reader.jsonData.message);
                } else {
                    if(!silentMode)
                    Ext.popupMsg.msg("Success","Deleted Successfully");
                }
            }

            this.setSilentMode(false);
        }
    },


});